src/excmds.ts

```diff
@@ -883,9 +883,28 @@ export async function restart() {
  * @param filename The name the file should be saved as.
  */
 //#content
-export async function saveas(...filename: string[]) {
-    if (filename.length > 0) {
-        return Messaging.message("download_background", "downloadUrlAs", window.location.href, filename.join(" "))
+export async function saveas(...args: string[]) {
+    let overwrite = false
+    let cleanup = false
+    const uniqueArgs = new Set(args)
```

Comment:
This is going to cause crazy bugs: `:saveas ~/Documents/this is a silly name for a folder/` won't work.

We should use the `while()` and `args.shift()` technique that we do elsewhere (in `hint()` perhaps?)

Reply:
@bovine3dom 

Will try to have a look as time permits. Thanks.

For the record, it seems that we have multiple problems here:

1. We can't pass more than one space in file/directory names
     - This is likely lingering from before with the `filename.join(" ")` pattern
     
2. We are losing repeating sub-strings in file/directory names e.g. `~/Documents/this is a silly name for a folder` becomes `~/Documents/this is a silly name for folder`

Reply:
I can look myself if you're sick of this code : )

I only left a comment because I spotted it and didn't want to forget. Thanks for all your work on this!

Reply:
> We can't pass more than one space in file/directory names
> This is likely lingering from before with the filename.join(" ") pattern

Oh god, why did you have to spot this, this is a fundamental issue with the command line